<launch>
  <arg name="gui" default="true"/>

  <node name="static_tf_publisher_map_to_camera"
        pkg="tf" type="static_transform_publisher"
        args="-0.580 -1.485 1.289 -0.753 0.291 -0.217 0.548 /map /camera 10"/>

  <node name="polygon_array_publisher"
        pkg="jsk_recognition_utils" type="polygon_array_publisher.py"
        output="screen">
    <rosparam>
      frame_id: camera
      polygons:
      - points:
        - [1.5, -1.0, 2.0]
        - [-1.0, -1.0, 4.0]
        - [0.5,  1.0, 3.0]
        label: 1
        likelihood: 0.8
      - points:
        - [1.5, -1.0, 4.0]
        - [-1.0, -0.5, 3.0]
        - [-2.5, 1.5, 1.0]
        - [0.0,  1.0, 2.0]
        label: 2
        likelihood: 0.6
      - points:
        - [0.0, -1.0, 2.0]
        - [-2.0, -1.0, 4.0]
        - [-0.6, 0.2, 1.0]
        label: 1
        likelihood: 0.8
    </rosparam>
  </node>

  <node name="camera_info_publisher"
        pkg="rostopic" type="rostopic"
        args="pub -r 30 -s /camera/rgb/camera_info sensor_msgs/CameraInfo
              '{header: {stamp: now, frame_id: camera},
                height: 1080,
                width: 1920,
                K: [1064.3187262543424, 0.0, 960.0, 0.0, 1064.3981464626195, 540.0, 0.0, 0.0, 1.0],
                R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
                P: [1064.3187262543424, 0.0, 960.0, 0.0, 0.0, 1064.3981464626195, 540.0, 0.0, 0.0, 0.0, 1.0, 0.0]}'"/>

  <node name="polygon_array_to_label_image"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_perception/PolygonArrayToLabelImage">
    <remap from="~input" to="polygon_array_publisher/output"/>
    <remap from="~input/camera_info" to="camera/rgb/camera_info"/>
  </node>

  <group if="$(arg gui)">
    <node name="label_image_decomposer"
          pkg="jsk_perception" type="label_image_decomposer.py">
      <remap from="~input/label" to="polygon_array_to_label_image/output"/>
      <rosparam>
        only_label: true
        bg_label: 0
      </rosparam>
    </node>

    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_perception)/sample/config/sample_polygon_array_to_label_image.rviz"/>
  </group>
</launch>
